src/ngx_http_lua_socket_tcp.c

```diff
@@ -2751,8 +2760,26 @@ ngx_http_lua_socket_tcp_settimeouts(lua_State *L)
     }
 
     connect_timeout = (ngx_int_t) lua_tonumber(L, 2);
+    if (connect_timeout > NGX_HTTP_LUA_SOCKET_MAX_TIMEOUT) {
+        ngx_log_error(NGX_LOG_ERR, ngx_cycle->log, 0,
+                      "lua tcp socket connect timeout %i will overflow, "
+                      "use former setting instead", connect_timeout);
+        goto timeout_overflow;
+    }
     send_timeout = (ngx_int_t) lua_tonumber(L, 3);
```

Comment:
Style: please always add a blank line after `if` code blocks. Please fix other similar places as well. Thank you!